File: problems/75/solution.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int lo = 0, mid = 0, hi = size(nums) - 1;
        
        while (mid <= hi) {
            if (nums[mid] == 0) {
                swap(nums[mid++], nums[lo++]);
            } else if (nums[mid] == 2) {
                swap(nums[mid], nums[hi--]);
            } else {
                mid++;
            }
        }
    }
};

File: problems/75/solution.go
package main
func sortColors(nums []int)  {
    p0, p2, cur := 0, len(nums) - 1, 0
    
    for cur <= p2 {
        if nums[cur] == 0 {
            nums[cur], nums[p0] = nums[p0], nums[cur]
            cur++
            p0++
        } else if nums[cur] == 2 {
            nums[cur], nums[p2] = nums[p2], nums[cur]
            p2--
        } else {
            cur++
        }
    }
}